lib/plugins/entities.js

```diff
@@ -374,7 +374,8 @@ function inject (bot) {
 
   bot._client.on('damage_event', (packet) => { // 1.20+
     const entity = bot.entities[packet.entityId]
-    bot.emit('entityHurt', entity)
+    const source = bot.entities[packet.sourceCauseId - 1] // damage_event : SourceCauseId : The ID + 1 of the entity responsible for the damage, if present. If not present, the value is 0
+    bot.emit('entityHurt', entity, source)
```

Comment:
source will be empty for all old versions

Reply:
well of course. as the comment said for that event. 1.20+

Reply:
bot._client.on('damage_event', (packet) => { // 1.20+ <--- right here. this was already there before i added my code.